htdocs/categories/class/categorie.class.php

```diff
@@ -1957,4 +1957,97 @@ public static function replaceThirdparty(DoliDB $db, $origin_id, $dest_id)
 
         return CommonObject::commonReplaceThirdparty($db, $origin_id, $dest_id, $tables, 1);
     }
+
+	/**
+	 * Return a list with all selected categories of a category filter box
+	 *
+	 * @param string	$type			The category type (e.g Categorie::TYPE_WAREHOUSE)
+	 * @return Array					A list with all selected categories (Note: "-2" is typical "without category")
+	 */
+	public static function getPost($type)
```

Comment:
An  object.class.php file contains code that can be executed into a server only context so it should never contains GETPOST call.
I think you can move this line back into the list.php